### DISCUSSION TITLE
What is the best way to style the new inlay type hints in the Dark Plus theme?

### DISCUSSION THREAD

**Discussion by David-Else** [helix-editor/helix#6270] [General]
@kirawi @janos-r @AlexanderBrevig @korken89 

Now we have type hints! https://github.com/helix-editor/helix/pull/5934. This is how VS Code looks (I think, this screen shot from the docs could be out of date):

## VS Code

![inlay-hints](https://user-images.githubusercontent.com/12832280/221609262-16c60118-3b17-44b7-8c91-bac25c7baa14.png)

Dark Plus styles them in white at the moment as it has no `"ui.virtual.inlay-hint" ` yet, and that doesn't look great as they really need to either stand out or be a lot fainter so they are obviously different from the rest of the code:

## Right now, no `"ui.virtual.inlay-hint"

![Screenshot from 2023-02-27 15-43-48](https://user-images.githubusercontent.com/12832280/221610288-20049d9d-a854-4b9a-a846-2c6787dcfc6a.png)

This is how it would look with `"ui.virtual.inlay-hint" = {fg = "text", bg = "dark_gray4" }`. This seems the closest match to VS Code, but I might be wrong. It doesn't look great to me, needs space at the edges and the original rounding looks much better, what do people think?:

##  {fg = "text", bg = "dark_gray4" }`

![Screenshot from 2023-02-27 16-58-06](https://user-images.githubusercontent.com/12832280/221629370-eb51e9f3-5f37-4af0-b76d-615a962350e5.png)

maybe just `fg = "dark_gray4"`, not following VS Code, but minimal is probably best as it makes them less distracting. What are the other themes doing?:

## `fg = "dark_gray4"`

![Screenshot from 2023-02-27 17-45-42](https://user-images.githubusercontent.com/12832280/221642797-31ab9880-12b1-41b6-90d2-aafcdc589787.png)

**Comment by korken89** [comment]
Awesome to hear! IMO the last suggestion is the best, minimal is more :)

**Reply by korken89** [reply]
Alright, I have tested the last suggestion on a good screen (XPS15) and an old screen Thinkpad X230, and it works well on the XPS15 but disappears on older screens with less contrast unfortunately.

**Comment by ivtkac** [comment]
What do you think about `{ fg = "white", bg = "#424242" }`, where `#424242` is blending colors `text` and `background` by 10%.

How it looks in helix:
![image](https://user-images.githubusercontent.com/116971836/224483963-d0e2a673-3e4e-463a-96e6-643ed61304c0.png)


and original in vscode:
![image](https://user-images.githubusercontent.com/116971836/224483927-0678fb37-239f-4c63-8ea3-41d981cfc508.png)

**Reply by korken89** [reply]
Looks nice!

**Reply by korken89** [reply]
I did also test this on a good screen (XPS15) and an old screen Thinkpad X230, and it works well on both.

**Comment by David-Else** [comment]
I had a look with the browser developer tools and the background grey in your VS Code screenshot  is `#444444`:

![Screenshot from 2023-03-11 13-48-25](https://user-images.githubusercontent.com/12832280/224488452-466bd20a-9f6c-4974-82fd-dd14fff8af6c.png)

It is hard to grab the white, but I have a feeling that `"white"` is too bright, can we look at the original VS Code files and get an exact match? What do you think of `fg = "text"`?

**Reply by ivtkac** [reply]
I agree `fg = "text"` is reasonable.

**Comment by ivtkac** [comment]
Looking at https://github.com/microsoft/vscode/blob/cd1d31179d612c380262746f00d35ed072dc7728/build/lib/stylelint/vscode-known-variables.json#L106-L111, I found out the VS Code uses `#ffffff` and  `rgba(77, 77, 77, 0.8)`. As Helix doesn't support opacity to color (I'm not sure), we could use `#4d4d4d` (same as original background color in VS Code just without opacity).

![зображення](https://user-images.githubusercontent.com/116971836/224489538-9cce9864-f3eb-4813-9a83-ac989c56c8a9.png)

And I agree `white` is too bright, so `fg = "text"` looks good to me.

By the way, we could also copy the [dark+ experimental version](https://github.com/microsoft/vscode/blob/cd1d31179d612c380262746f00d35ed072dc7728/extensions/theme-defaults/themes/dark_plus_experimental.json#L43-L46) port, they use `#8b949e` and `#8b949e33` (similar to `#353739`)

How it looks:

![зображення](https://user-images.githubusercontent.com/116971836/224490467-f83b8f0a-69e9-4250-ac74-b62fadf5ad84.png)

**Reply by David-Else** [reply]
I think the [dark+ experimental version](https://github.com/microsoft/vscode/blob/cd1d31179d612c380262746f00d35ed072dc7728/extensions/theme-defaults/themes/dark_plus_experimental.json#L43-L46)  inlay hints look the best, what exactly did you enter to get that screenshot? #8b949e33 and #8b949e give funny results, sorry I don't know anything about colour codes if that is a dumb question.

**Reply by gabydd** [reply]
#8b949e33 is hex code plus an alpha(transparency) value which seems to be about equivalent too #E8EAEC
